### DISCUSSION TITLE
Is it possible to create a hybrid mesh sharding of GPU and CPU?

### DISCUSSION THREAD

**Discussion by nasyxx** [jax-ml/jax#18372] [Q&A]
I have tried

` sharding = PositionalSharding(mesh_utils.create_device_mesh((3,), jax.devices() + jax.devices("cpu")))`

<img width="1113" alt="Screenshot 2023-11-03 at 10 48 25 AM" src="https://github.com/google/jax/assets/5007613/e52d96d0-601b-4297-ab76-a2c1ee6c9e7c">

However, JAX regarded the CPU as the GPU. At the same time, I couldn't put the array on it.

In hugging face, it is possible to put the model on GPU and CPU using `device_map`.  Is it possible to do this in JAX?

**Comment by Kelvinlby** [comment]
+1

**Comment by yashk2810** [comment]
Not really. But if you describe what you are trying to do, I can probably suggest alternatives.
